### PR TITLE
Changed code to fit repository style and fixed wrong CLI application name

### PR DESCRIPTION
**Pull request by zillemarco**
Changed code to fit repository style and fixed wrong CLI application name (see #745)

### CODE REVIEW

File: src/CLI/Generator.cs [hunk 1]
```diff
@@ -6,155 +6,158 @@
...
 
 namespace CppSharp
 {
     class Generator : ILibrary
     {
-        private Options _options;
-
-        private String _triple = "";
-        private CppAbi _abi = CppAbi.Microsoft;
+        private Options Options = null;
```
Comment (ddobrev) [inline src/CLI/Generator.cs]: Private fields start with a lowercase letter. To make a distinction between a field and a parameter, use "this":

this.options = options;

File: src/CLI/Options.cs [hunk 2]
```diff
@@ -16,52 +11,36 @@ enum TargetArchitecture
...
-        public List<String> Libraries { get { return _libraries; } set { _libraries = value; } }
+        public List<string> Libraries { get; } = new List<string>();
 
-        private Dictionary<String, String> _defines = new Dictionary<String, String>();
-        public Dictionary<String, String> Defines { get { return _defines; } set { _defines = value; } }
+        public Dictionary<string, string> Defines { get; } = new Dictionary<string, string>();
 
-        private String _outputDir = "";
-        public String OutputDir { get { return _outputDir; } set { _outputDir = value; } }
+        public string OutputDir { get; set; } = "";
```
Comment (ddobrev) [inline src/CLI/Options.cs]: Now that you use string.IsNullOrEmpty, this initialisation is needless, the default value of "null" will work just the same:

public string OutputDir { get; set; };

File: src/CLI/Options.cs [hunk 3]
```diff
@@ -16,52 +11,36 @@ enum TargetArchitecture
...
-        public TargetArchitecture Architecture { get { return _architecture; } set { _architecture = value; } }
+        public TargetArchitecture Architecture { get; set; } = TargetArchitecture.x86;
 
-        private GeneratorKind _kind = GeneratorKind.CSharp;
-        public GeneratorKind Kind { get { return _kind; } set { _kind = value; } }
+        public GeneratorKind Kind { get; set; } = GeneratorKind.CSharp;
         
-        private bool _checkSymbols = false;
-        public bool CheckSymbols { get { return _checkSymbols; } set { _checkSymbols = value; } }
+        public bool CheckSymbols { get; set; } = false;
```
Comment (ddobrev) [inline src/CLI/Options.cs]: Boolean fields are false by default, no need to initialise.

File: src/CLI/Generator.cs [hunk 4]
```diff
@@ -13,151 +13,151 @@ namespace CppSharp
...
         {
             var parserOptions = driver.ParserOptions;
-            parserOptions.TargetTriple = Triple;
-            parserOptions.Abi = Abi;
+            parserOptions.TargetTriple = triple;
+            parserOptions.Abi = abi;
 
             var options = driver.Options;
-            options.LibraryName = Options.OutputFileName;
+            options.LibraryName = this.options.OutputFileName;
```
Comment (ddobrev) [inline src/CLI/Generator.cs]: We only use "this" when we have to make a distinction, for example in a constructor.

File: src/CLI/Generator.cs [hunk 5]
```diff
@@ -6,155 +6,158 @@
...
         {
             var parserOptions = driver.ParserOptions;
-            parserOptions.TargetTriple = _triple;
-            parserOptions.Abi = _abi;
+            parserOptions.TargetTriple = triple;
+            parserOptions.Abi = abi;
 
             var options = driver.Options;
-            options.LibraryName = _options.OutputFileName;
+            options.LibraryName = this.options.OutputFileName;
```
Comment (ddobrev) [inline src/CLI/Generator.cs]: Some more "this" to remove. To make sure, please search for all usages of the field.

**Comment by zillemarco** [comment]
I've made the changes and squashed them together.

**Comment by ddobrev** [comment]
@zillemarco I am afraid I have to ask you to send your pull request again. gayhump obliterated our history by adding my comment to the commit message instead of the discussion here as it had always done before. I am really sorry about this but I had no way of expecting such a moronic behaviour.
Now that you have to do it anyway, please separate just the change for renaming the application into its own commit. It has no logical connection to your refactoring so for better history it really should have its own record.

**Comment by zillemarco** [comment]
All right, no problem. So I just need to divide the commit into two again and then redo the pull request if I understood correctly?

**Comment by ddobrev** [comment]
Right.

**Pull request merged**